### ISSUE TITLE
Null Safety issue on the given example!

### ISSUE THREAD

**Issue by kishan-dhankecha** [flutter/website#11545]
### Page URL

https://docs.flutter.dev/cookbook/effects/shimmer-loading/

### Page source

https://github.com/flutter/website/tree/main/src/content/cookbook/effects/shimmer-loading.md

### Describe the problem

There is an issue with the code provided in the example below. When I copied the whole code from the interactive editor in the website and pasted it into my project to use the `ShimmerLoading` widget, I found that I get the classic `Null check operator used on a null value` error. See the issue in the below snippet.

```dart
// Collect ancestor shimmer information.
final shimmer = Shimmer.of(context)!; /// Error comes from this line
if (!shimmer.isSized) {
  // The ancestor Shimmer widget isn't laid
  // out yet. Return an empty box.
  return const SizedBox();
}
```

### Expected fix

It should be changed to this,

```dart
// Collect ancestor shimmer info.
final shimmer = Shimmer.of(context);
if (shimmer == null || !shimmer.isSized) {
  // The ancestor Shimmer widget has not laid
  // itself out yet. Return an empty box.
  return const SizedBox();
}
```

### I would like to fix this problem.

- [X] I will try and fix this problem on docs.flutter.dev.

**Comment by sfshaza2** [comment]
@kishan-dhankecha, what version of Flutter/Dart are you using? `shimmer` is defined as never being null. So, testing for null isn't necessary. The only thing that makes sense is that you are using an older version of Dart.

**Comment by kishan-dhankecha** [comment]
Note that this issue of null check only happens when a user forgets to wrap the widget with `Shimmer`. In that case, context will not find the `Shimmer` on the upper side of the tree.

**Comment by sfshaza2** [comment]
@domesticmouse, can you weigh in on this?

**Comment by domesticmouse** [comment]
Given this is an exception that is only thrown when the developer goofs, I don't see the need to make the suggested change. I'd prefer the code to break, given it is already broken.

**Comment by kishan-dhankecha** [comment]
Yes, @domesticmouse you are right here. It would help more if the developer discovered this if they made this mistake.

I am closing this. Thanks.
